oshdb-api/src/main/java/org/heigit/bigspatialdata/oshdb/api/mapreducer/MapReducer.java

```diff
@@ -586,43 +586,51 @@ protected MapReducer(MapReducer<?> obj) {
 
   /**
    * Adds an osm tag filter: The analysis will be restricted to osm entities that have at least one
-   * of the supplied tags (key=value pairs).
+   * of the supplied tags (key=value pairs or key=*).
    *
-   * @param tags the tags (key/value pairs) to filter the osm entities for
+   * @param tags the tags (key/value pairs or key=*) to filter the osm entities for
    * @return a modified copy of this mapReducer (can be used to chain multiple commands together)
    */
   @Contract(pure = true)
-  public MapReducer<X> osmTag(Collection<OSMTag> tags) {
+  public MapReducer<X> osmTag(Collection<? extends OSMTagInterface> tags) {
     MapReducer<X> ret = this.copy();
-    if (tags.size() == 0) {
+    if (tags.isEmpty()) {
       LOG.warn("Empty tag list. No data will match this filter.");
       ret.preFilters.add(ignored -> false);
       ret.filters.add(ignored -> false);
       return ret;
     }
+    Set<Integer> preKeyIds = new HashSet<>();
```

Comment:
we might add a comment here as to what this Set is about. It took me a while to get the difference to the keyIds